apps/opencs/view/render/terrainshapemode.cpp

```diff
@@ -0,0 +1,1290 @@
+#include "terrainshapemode.hpp"
+
+#include <algorithm>
+#include <string>
+#include <sstream>
+#include <memory>
+
+#include <QWidget>
+#include <QIcon>
+#include <QEvent>
+#include <QDropEvent>
+#include <QDragEnterEvent>
+#include <QDrag>
+
+#include <osg/Group>
+#include <osg/Vec3f>
+
+#include <components/esm/loadland.hpp>
+#include <components/debug/debuglog.hpp>
+
+#include "../widget/brushshapes.hpp"
+#include "../widget/modebutton.hpp"
+#include "../widget/scenetoolbar.hpp"
+#include "../widget/scenetoolshapebrush.hpp"
+
+#include "../../model/doc/document.hpp"
+#include "../../model/prefs/state.hpp"
+#include "../../model/world/columnbase.hpp"
+#include "../../model/world/commandmacro.hpp"
+#include "../../model/world/commands.hpp"
+#include "../../model/world/data.hpp"
+#include "../../model/world/idtable.hpp"
+#include "../../model/world/idtree.hpp"
+#include "../../model/world/land.hpp"
+#include "../../model/world/resourcetable.hpp"
+#include "../../model/world/tablemimedata.hpp"
+#include "../../model/world/universalid.hpp"
+
+#include "editmode.hpp"
+#include "pagedworldspacewidget.hpp"
+#include "mask.hpp"
+#include "object.hpp" // Something small needed regarding pointers from here ()
+#include "terrainselection.hpp"
+#include "worldspacewidget.hpp"
+
+CSVRender::TerrainShapeMode::TerrainShapeMode (WorldspaceWidget *worldspaceWidget, osg::Group* parentNode, QWidget *parent)
+: EditMode (worldspaceWidget, QIcon {":scenetoolbar/editing-terrain-shape"}, Mask_Terrain | Mask_Reference, "Terrain land editing", parent),
+    mBrushSize(1),
+    mBrushShape(CSVWidget::BrushShape_Point),
+    mShapeBrushScenetool(0),
+    mDragMode(InteractionType_None),
+    mParentNode(parentNode),
+    mIsEditing(false),
+    mTotalDiffY(0),
+    mShapeEditTool(ShapeEditTool_Drag),
+    mShapeEditToolStrength(8),
+    mTargetHeight(0)
```

Comment:
You can put all default initialization into class definition like:
```c++
int mTargetHeight = 0;
```
Here are guidelines about this: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c48-prefer-in-class-initializers-to-member-initializers-in-constructors-for-constant-initializers .

Reply:
This was deliberate, as much of the OpenMW-CS code follows the convention of using member initializer list instead of class definition. Changeable of course, but I see no clear advantages when compared to member initializer lists. If this is more of a preference than a rule, I'd rather keep the uniformity. Personally at least I feel it's easier to work with code when it mostly follows the same formatting.

Reply:
I'll let this be here for a while, hoping for a second opinion. The point of changing to in class initializers stands, but I'm thinking whether to go with code uniformity for this PR, then later change the whole OpenMW-CS codebase to follow this guideline in one bigger PR. (while waiting, I'll prepare another PR to improve the code of terrain texture editing that's already merged, but there are similar issues that this PR had)